src/contexts/useAuth.jsx

```diff
@@ -21,9 +34,27 @@ const AuthProvider = ({children}) => {
         return auth.signOut()
     }
 
+    const setCurrentUser = (uid, data) => {
+        setUser({
+            uid: uid,
+            first_name: data.first_name,
+            last_name: data.last_name,
+            email: data.email,
+            plan: data.plan,
+            stripe_id: data.stripe_id,
+            watchlist: data.watchlist
+        })
+    }
+
     useEffect(() => {
         return auth.onAuthStateChanged(firebaseUser => {
-            setUser(firebaseUser)
+            if (firebaseUser != null) {
```

Comment:
I am not sure why this is all under return

Reply:
fixed in commit "fixed useEffect return, combined user name..."